processing/src/main/java/io/druid/query/MetricsEmittingQueryRunner.java

```diff
@@ -85,21 +84,13 @@ public MetricsEmittingQueryRunner(
   public Sequence<T> run(final Query<T> query, final Map<String, Object> responseContext)
   {
     final QueryMetrics<? super Query<T>> queryMetrics = queryToolChest.makeMetrics(query);
-
     applyCustomDimensions.accept(queryMetrics);
-
+    final Query<T> queryWithMetrics = query.withQueryMetrics(queryMetrics);
```

Comment:
same question, does this still need to be done if the query already has a `QueryMetrics` set in it?